### PR TITLE
Add simple profiling tests

### PR DESCRIPTION
**Pull request by nsavoire**
# What does this PR do?

A brief description of the change being made with this pull request.

# Motivation

What inspired you to submit this pull request?  In order to motivate your point, it may be valuable to explain why this change is useful or what problem is being solved.

# Additional Notes

This is the section to put technical guidance/constraints, call out potential regressions, cite sources, and in general offer some exposition on the _how_ of the PR.  This section doesn't need to be incredibly detailed, but it makes life easier for reviewers!

# How to test the change?

Describe here in detail how the change can be validated.  This is a great section to call out specific tests you've added or improved, or to acknowledge code sections which are particularly difficult to test.

### CODE REVIEW

**Comment by nsavoire** [comment]
Current dependencies on/for this PR:
* main
  * **PR #141** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/141" target="_blank">Graphite</a> 
    * **PR #142** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/142" target="_blank">Graphite</a> 
      * **PR #143** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/143" target="_blank">Graphite</a> 
        * **PR #144** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/144" target="_blank">Graphite</a> 
          * **PR #145** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/145" target="_blank">Graphite</a> 
            * **PR #146** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/146" target="_blank">Graphite</a> 
              * **PR #147** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/147" target="_blank">Graphite</a> 
                * **PR #148** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/148" target="_blank">Graphite</a> 
                  * **PR #149** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/149" target="_blank">Graphite</a> 
                    * **PR #150** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/150" target="_blank">Graphite</a> 
                      * **PR #151** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/151" target="_blank">Graphite</a> 
                        * **PR #152** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/152" target="_blank">Graphite</a> 
                          * **PR #153** <a href="https://app.graphite.dev/github/pr/DataDog/ddprof/153" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/ddprof/153?utm_source=stack-comment).

File: test/simple_malloc.cc [hunk 1]
```diff
@@ -3,20 +3,186 @@
...
+        break;
+      }
+    }
+
+    if (!exec_args.empty()) {
+      std::vector<char *> new_args;
+      for (auto &a : exec_args) {
+        new_args.push_back(a.data());
+      }
+      execvp(new_args[0], new_args.data());
```
Comment (r1viollet) [inline test/simple_malloc.cc:164]: Interesting mechanism 👍

File: test/simple_malloc.cc [hunk 2]
```diff
@@ -3,20 +3,186 @@
...
+
+    unsigned int nb_forks{1};
+    unsigned int nb_threads{1};
+    unsigned int sleep_us{0};
+    unsigned int spin_us{0};
+    int timeout_ms = -1;
+    uint64_t loop_count = std::numeric_limits<uint64_t>::max();
+    std::vector<std::string> exec_args;
+
+    app.add_option("--fork", nb_forks, "Number of processes to create")
```
Comment (r1viollet) [inline test/simple_malloc.cc:120]: `simple` multithreaded, multi-process that can be arranged to run custom workloads 😄

**Review by r1viollet — APPROVED** [review]
LGTM

**Comment by nsavoire** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/ddprof/153?utm_source=stack-comment).


Job Reference: 9lylUEXBAvDRoXRafmxH

**Pull request merged**